addons/im_livechat/models/discuss_channel_member.py

```diff
@@ -40,17 +46,28 @@ def create(self, vals_list):
             member.sudo().livechat_member_type = "agent"
         return members
 
-    def _compute_livechat_member_type(self):
+    def _compute_from_history(self):
         for member in self:
             member.livechat_member_type = member.livechat_member_history_ids.livechat_member_type
+            member.chatbot_script_id = member.livechat_member_history_ids.chatbot_script_id
 
-    def _inverse_livechat_member_type(self):
+    def _create_history_if_not_exists(self):
```

Comment:
Wouldn't we want to give a map of member id -> fields so we can create everything correctly rather than requiring a separate write afterwards?